manifests/charts/istio-control/istio-discovery/templates/reader-clusterrole.yaml

```diff
@@ -21,6 +21,9 @@ rules:
   - apiGroups: ["networking.istio.io"]
     verbs: [ "get", "watch", "list" ]
     resources: [ "workloadentries" ]
+  - apiGroups: ["networking.x-k8s.io", "gateway.networking.k8s.io"]
+    resources: ["gateways"]
```

Comment:
This new dependency may be tricky. You mentioned when we don't have RBAC, everything is blocked? Is that right? Or at best it times out after 15s?

Do we have users always update this first before the control plane? I would think we don't?

Reply:
You'lll get 0 endpoints from that cluster. I have the feature flag off for now. 

I think if the flag is off by default for a few versions it should be ~safe enough. Skip version upgrades might have an issue though.